### ISSUE TITLE
`lmfit` ignores `wt` parameter

### ISSUE THREAD

**Issue by jw3126** [JuliaNLSolvers/LsqFit.jl#129]
The `lmfit` function does not propagate the `wt` parameter to `levenberg_marquardt`.
https://github.com/JuliaNLSolvers/LsqFit.jl/blob/8b163b79f4871cdac416943a17a1508d304ffdf4/src/curve_fit.jl#L64

**Comment by jw3126** [comment]
Also the `autodiff` parameter is not used.

**Comment by dm3** [comment]
The weight parameter is applied in the `curve_fit` and AFAIK passed through to `lmfit` to be used in the `LsqFitResult`. The `autodiff` parameter could probably be removed as it's unused.

**Comment by Magalame** [comment]
Yep, this isn't really an issue, more of a "coding style" concern

**Comment by pkofod** [comment]
I agree the autodiff should be removed. `lmfit` is really superfluous as well to be honest, but... I'm going to be using another LM implementation in the near future, so we'll just remove `lmfit` then.

**Comment by pkofod** [comment]
I agree, but now I think it should just accept all the options as some kind of options type and kill all `kwargs...` as the often make things hard to debug. autodiff should not be there and neither should the weight.
